File: learn-to-compile-scala-another-language/learn-about-compilers-scala/src/main/scala/tac/tac.scala
package tac // package is like a name space 

abstract class TACInstr

// Class definition also servers as the primary constructor
// in scala var (mutable) val (immutable)
case class TACOp(var tacTempValue: String, var op: String, var arg1: String = "", var arg2: String = "") extends TACInstr
{

}

case class TACConst(var tacTempValue: String, var value: String) extends TACInstr 
{

}

class TACTempGenerator 
{
    private var _count: Int = 0

    def NewTacInstr(): String =
    {
        _count += 1
        return s"t${_count}"
    }
} 

File: learn-to-compile-bytecode-go/tac.go
package main

type TacExpression struct {
	Type         string `json:"$type"`
	TacTempValue string `json:"tacTempValue"`
	Value        string `json:"value"`
	Op           string `json:"op"`
	Arg1         string `json:"arg1"`
	Arg2         string `json:"arg2"`
}

type Tac struct {
	Opcode string        `json:"opcode"`
	Tacvar TacExpression `json:"tacvar"`
}
